src/main/java/com/mathworks/ci/MatlabTestFile.java

```diff
@@ -0,0 +1,95 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2024, The MathWorks Inc.
+ *
+ * Class to store MATLAB test file information
+ * 
+ */
+
+import java.util.List;
+import java.util.ArrayList;
+
+import org.apache.commons.lang.RandomStringUtils;
+
+public class MatlabTestFile {
+    private String path;
+    private String name;
+    private Double duration;
+    private String status;
+    private List<MatlabTestCase> matlabTestCases;
+    private String id;
+
+    public MatlabTestFile() {
+        this.path = "";
+        this.name = "";
+        this.duration = 0.0;
+        this.status = MatlabBuilderConstants.NOT_RUN;
+        this.matlabTestCases = new ArrayList<MatlabTestCase>();
+        this.id = RandomStringUtils.randomAlphanumeric(8);
+    }
+
+    public void incrementDuration(Double matlabTestCaseDuration) {
+        this.duration += matlabTestCaseDuration;
+    }
+
+    public void updateStatus(MatlabTestCase matlabTestCase) {
+        if (!this.status.equals(MatlabBuilderConstants.FAILED)) {
+            if (matlabTestCase.getFailed()){
+                this.status = MatlabBuilderConstants.FAILED;
+            }
+            else if (!this.status.equals(MatlabBuilderConstants.INCOMPLETE)){
+                if (matlabTestCase.getIncomplete()){
+                    this.status = MatlabBuilderConstants.INCOMPLETE;
+                }
+                else if (matlabTestCase.getPassed()){
+                    this.status = MatlabBuilderConstants.PASSED;
+                }
+            }
+        }
+    }

```

Comment:
Seems a little strange here too with the `updateStatus` method because it only uses the boolean properties of the MatlabTestCase and not the status. It looks like we update the MatlabTestFile status as each MatlabTestCase comes in.